### PR TITLE
DRY of type check and len check code in `compute`

### PR DESCRIPTION
**Pull request by yjhmelody**

### CODE REVIEW

File: src/compute/arithmetics/basic/common.rs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+use crate::array::{Array, PrimitiveArray};
+use crate::error::{ArrowError, Result};
+use crate::types::NativeType;
+
+// Checking if both arrays have the same type
+#[inline]
+pub fn check_same_type<L: NativeType, R: NativeType>(
```
Comment (jorgecarleitao) [inline src/compute/arithmetics/basic/common.rs:7]:
```suggestion
pub(crate) fn check_same_type<L: NativeType, R: NativeType>(
```

File: src/compute/arithmetics/basic/common.rs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        return Err(ArrowError::InvalidArgumentError(
+            "Arrays must have the same logical type".to_string(),
+        ));
+    }
+    Ok(())
+}
+
+// Checking if both arrays have the same length
+#[inline]
+pub fn check_same_len<L: NativeType, R: NativeType>(
```
Comment (jorgecarleitao) [inline src/compute/arithmetics/basic/common.rs:21]:
```suggestion
pub(crate) fn check_same_len<L: NativeType, R: NativeType>(
```

**Review by jorgecarleitao — APPROVED** [review]
Thank you for your PR! Looks great. I left two suggestions so that we do not introduce any new public API.

**Comment by yjhmelody** [comment]
@jorgecarleitao i have add pub（crate）use common: : * already

**Pull request merged**